lib/puppet/util/log.rb

```diff
@@ -147,7 +147,12 @@ def Log.newdestination(dest)
       Puppet.log_exception(detail)
 
       # If this was our only destination, then add the console back in.
-      newdestination(:console) if @destinations.empty? and (dest != :console and dest != "console")
+      if destinations.empty? && (dest != :console)
```

Comment:
Since `newdestination` can be called from custom applications in modules, eg https://github.com/puppetlabs/puppetlabs-panos/blob/6edf9c652f35918eea3288aead1da03abb858c42/tasks/store_config.rb#L14, I think we need to preserve the behavior of allowing `dest` to be a string, like `dest.to_s != "console"` or `dest.intern != :console`